src/populateExchange.ts

```diff
@@ -86,13 +88,10 @@ export const populateExchange = ({
       query,
     });
 
-    userFragments = newFragments.reduce(
-      (state, fragment) => ({
-        ...state,
-        [fragment.name.value]: fragment,
-      }),
-      userFragments
-    );
+    userFragments = newFragments.reduce((state, fragment) => {
```

Comment:
Since we are effectively just mutating the object I'd convert this to

```
newFragments.forEach(frag => {
  userFragments[getName(frag)] = frag
})
```

Diminishes the need for a `return` keyword and is a bit more readable since we're mutating anyway